cpp/src/umap/runner.h

```diff
@@ -69,22 +69,22 @@ __global__ void init_transform(int *indices, T *weights, int n,
 }
 
 /**
-     * Simple helper function to set the values of an array to zero.
-     *
-     * @param vals array of values
-     * @param nnz size of array of values
-     */
+ * Simple helper function to set the values of an array to zero.
+ *
+ * @param vals array of values
+ * @param nnz size of array of values
+ */
 template <int TPB_X>
 __global__ void reset_vals(int *vals, int nnz) {
```

Comment:
Rather writing custom kernels, you should can ml-prims for these types of simple operations. For instance unary_op function of linalg is pretty good fit for this functions. 

https://github.com/rapidsai/cuml/blob/branch-0.10/cpp/src_prims/linalg/unary_op.h

Reply:
You may check all the custom kernels whether you can use any available prims or not.

Reply:
Agreed. Will use unary op for this one.